docs/manifests.rst

```diff
@@ -3,46 +3,55 @@ Manifests
-    ]
-  }
+This page provides information and guidance on how to use manifests, including an explanation of the most common parameters that can be specified. It is recommended to have followed the :doc:`first-build` tutorial before reading this section, and to be familiar with :doc:`flatpak-builder`.
+
+Manifest files should be named using the application ID. For example, the manifest file for GNOME Dictionary is named ``org.gnome.Dictionary.json``. This page uses this manifest file, which was introduced in :doc:`first-build`, for all its examples.
+
+A complete list of all the properties that can be specified in manifest files can be found in the :doc:`flatpak-builder-command-reference`, as well as the ``flatpak-builder`` man page.
+
+Basic parameters
+----------------
+
+Each manifest file should specify basic information about the application that is to be built, including the ``app-id``, ``runtime``, ``runtime-version``, ``sdk`` and ``command`` parameters. These properties are typically specified at the beginning of the file.
 
-As can be seen, this manifest includes basic information about the application before specifying a single .tar file to be downloaded and built. More complex manifests include a sequence of modules. Module sources can be of several types, including ``.tar`` or ``.zip`` archives, Git or Bzr repositories, patch files or shell commands that are run.
+For example, the GNOME Dictionary manifest includes::
 
-Each of the properties that can be specified in a manifest file are listed in the :doc:`flatpak-builder-command-reference`, as well as the ``flatpak-builder`` man page.
+  "app-id": "org.gnome.Dictionary",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.26",
+  "sdk": "org.gnome.Sdk",
+  "command": "gnome-dictionary",
 
-.. TODO
+Specifying a runtime and runtime version allows that the runtime that is needed by your application to be automatically installed on users' systems.
 
-  Add a section on the top level options, possibly called "Application metadata". Ideally this should recommend a basic set of options that developers should start with.
+File renaming
+-------------
+
+As was described in :doc:`introduction`, exports are application files that are made available to the host, and include things like the application's ``.desktop`` file and icon.
+
+The names of files that are exported by a Flatpak must prefixed using the application ID, such as ``org.gnome.Dictionary.desktop``. The easiest way to do this is to use these file names in the application's source file.
+
+However, if this is not possible, ``flatpak-builder`` allows the exports to be renamed as part of the build process, by specifying one of the following properties in the manifest:
 
-  Add a section on specifying the list of modules. There has to be something to say here...
+- ``rename-icon`` - rename the application icon
+- ``rename-desktop-file`` - rename the ``.desktop`` filename
+- ``rename-appdata-file`` - rename the AppData file
```

Comment:
You should note that these should not be relied upon by upstreams and they should always rename their files there. That fixes subtle mistakes like icons being referenced internally and such.

Reply:
If that's the case, should I just remove the entire section on file renaming?

Reply:
They certainly need to be documented because a lot of applications use it, there just needs to be clear information on the better solution like #99

Reply:
Alright, I've just added a note for now.